### DISCUSSION TITLE
Recording CF data while manually piloting

### DISCUSSION THREAD

**Discussion by Itai51** [bitcraze/discussions#1411] [Q&A]
Hello CF community!

I was told in a [different discussion](https://github.com/orgs/bitcraze/discussions/1360#discussioncomment-9673834) that I cannot connect to the CF both via the client and via a script at the same time.

I would still like to record lighthouse data via a script, but I am not sure how. Obviously I cannot connect an input device (e.g. a game controller) via the client, but when I try the smartphone app, it also refuses to connect while a script is running.

I feel like there is a simple solution that I'm just missing, but how can I pilot my CF manually, while recording data in a script using the CF-lib?

Thank you!

**Comment by gemenerik** [comment]
You can write logs to file in the client. Set up your logs in `settings`>`logging configurations`. Set your `logging period` as desired. In the `Log Blocks` tab check `Write to file` for the desired logs. You can find the saved logs in `settings`>`open config folder`>`logdata folder`

Alternatives are writing data to [the micro-SD card deck](https://www.bitcraze.io/products/micro-sd-card-deck/), or implementing controller support directly into cflib.

**Reply by Itai51** [reply]
What you are describing is using the built-in 'log blocks' tab in the CF-client, correct? I am wondering if there is a way to run a script which I write using the cflib alongside manual controls. Perhaps the implementation of controller support in the cflib which you mentioned?

**Reply by gemenerik** [reply]
Yes that's correct!

If you want to combine script and manual control I think indeed it is best to add controller support to the cflib. Perhaps there are some Python libraries that can help you get started.
